### PR TITLE
kubectl debug: support different kinds

### PR DESCRIPTION
**Pull request by verb**
**What type of PR is this?**
/kind feature
/sig cli
/priority important-longterm

**What this PR does / why we need it**: This is a refactor ahead of adding support for supporting `kubectl debug` of nodes.

**Which issue(s) this PR fixes**:

WIP #89306

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://git.k8s.io/enhancements/keps/sig-cli/20190805-kubectl-debug.md#node-troubleshooting-with-privileged-containers
```

### CODE REVIEW

**Comment by verb** [comment]
Looks like that node-e2e-containerd test is failing for everyone.

/assign @soltysh

File: staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go [hunk 1]
```diff
@@ -209,38 +209,35 @@ func (o *DebugOptions) Validate(cmd *cobra.Command) error {
 
 // Run executes a kubectl debug.
 func (o *DebugOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
+	ctx := context.Background()
+
 	r := o.builder.
 		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
-		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.PodNames...).
+		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.TargetNames...).
```
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go:216]: With this in, you need to update the examples to use other resources, but that can be done in the next PR.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/retest

**Pull request merged**